src/main/java/org/zalando/nakadi/filters/LoggingFilter.java

```diff
@@ -34,47 +36,117 @@ public LoggingFilter(final NakadiKpiPublisher nakadiKpiPublisher,
         this.accessLogEventType = accessLogEventType;
     }
 
+    private class RequestLogInfo {
+
+        private String userAgent;
+        private String user;
+        private String method;
+        private String path;
+        private String query;
+        private String contentEncoding;
+        private Long contentLength;
+        private String acceptEncoding;
+
+        private RequestLogInfo(final HttpServletRequest request) {
+            this.userAgent = Optional.ofNullable(request.getHeader("User-Agent")).orElse("-");
+            this.user = Optional.ofNullable(request.getUserPrincipal()).map(Principal::getName).orElse("-");
+            this.method = request.getMethod();
+            this.path = request.getRequestURI();
+            this.query = Optional.ofNullable(request.getQueryString()).map(q -> "?" + q).orElse("");
+            this.contentEncoding = Optional.ofNullable(request.getHeader(HttpHeaders.CONTENT_ENCODING))
+                .orElse("-");
+            this.acceptEncoding = Optional.ofNullable(request.getHeader(HttpHeaders.ACCEPT_ENCODING))
+                .orElse("-");
+            this.contentLength = request.getContentLengthLong() == -1 ? 0 : request.getContentLengthLong();
+        }
+    }
+
+    private class AsyncRequestListener implements AsyncListener {
+        private final HttpServletRequest request;
+        private final HttpServletResponse response;
+        private long startTime;
+
+        private AsyncRequestListener(final HttpServletRequest request, final HttpServletResponse response,
+                                    final long startTime) {
+            this.request = request;
+            this.response = response;
+            this.startTime = startTime;
+
+            final RequestLogInfo requestLogInfo = new RequestLogInfo(request);
+            ACCESS_LOGGER.info("{} \"{}{}\" \"{}\" \"{}\" \"{}\" \"{}\"",
```

Comment:
Does it make more sense to log the stream parameters in the state machine? This will log twice for each request to '/subscriptions/{}/events'. I logged it here as we might need the user who made the request.

Reply:
I do not follow why you logged it here, the user is obtained in `RequestLogInfo`, so it will be logged in callback, right?

Reply:
Yes, it will be logged in the callback. We see the user who requested the stream when the stream is closed (in the case of 200). I wanted to ask is it important to see the user when the request is initiated. Or is it fine to log just the stream parameters in the state machine?